pages/inventories/keycaps.js

```diff
@@ -9,9 +13,52 @@ const StyledContainer = styled.div`
 `;
 
 export default function Keycaps() {
+  const [isOpen, setIsOpen] = useState(false);
+  const [userKeycaps, setUserKeycaps] = useState([]);
+  const userId = "guest_user";
+
+  //Fetch user keycaps
+  const { data, error, mutate } = useSWR(
+    `/api/inventories/userkeycaps?userId=${userId}`
+  );
+
+  useEffect(() => {
+    if (data) {
+      setUserKeycaps(data.map((keycap) => keycap._id));
+    }
+  }, [data]);
+
+  //Function for adding keycap ID to the userKeycaps array
+  const handleAddKeycap = async (keycapId) => {
+    if (!userKeycaps.includes(keycapId)) {
+      const updatedKeycaps = [...userKeycaps, keycapId];
+      setUserKeycaps(updatedKeycaps);
+
+      //Save the update in DB
+      await fetch("/api/inventories/userkeycaps", {
```

Comment:
```suggestion
const response = await fetch("/api/inventories/userkeycaps", {
```
So you can call `mutate()` (see commment below) after confirming the request was successful..